> Z tisíce sledovaných součástek typu XXX se každý rok v průměru K porouchá a je nahrazeno novými. Určete intenzitu poruch součástky λ v jednotkách $rok^{-1}$ a $hod^{-1}$.

In [1]:
# 1000 observed components
N = 1000

Intenzita poruch součástky v jednotkách $ \mathrm{rok^{-1}} $:

$$ \lambda_{c} = \frac{K}{N} = \frac{K}{1000}\ \mathrm{rok^{-1}} $$

In [2]:
def single_component_failure_intensity(K, N):
    return K / N

Intenzita poruch součástky v jednotkách $ \mathrm{hod^{-1}} $:
$$ \lambda_{ch} = \frac{\frac{K}{N}}{365 \cdot 24} \mathrm{hod^{-1}} = \frac{\frac{K}{1000}}{365 \cdot 24} \ \mathrm{hod^{-1}} $$

In [3]:
def single_component_failure_intensity_hours(K, N):
    return (K / N) / (365 * 24)

> Počítač se skládá z N součástek typu XXX. Porucha každé součástky znamená poruchu počítače. Pro počítač určete symbolicky (tj. jako vzoreček):
> - celkovou intenzitu poruch systému $λ_s$,
> - pravděpodobnost bezporuchového provozu $R(t)$,
> - s jakou pravděpodobností systém zrealizuje službu (tzv. misi, angl. mission) v době trvání T hodin.

Uvažujeme konstantní intenzitu poruchy jedné součástky počítače, tak poté platí, že celková intenzita poruchy počítače skládajícího se z $ N $ součástek je:

$$ \lambda_{s} = N \cdot \lambda_{c} $$

In [4]:
def computer_failure_intensity(component_count, component_failure_intensity):
    return component_count * component_failure_intensity

Dosazením intenzity poruchy počítače do vzorečku pro pravděpodobnost bezporuchového provozu ($ R(t) $) získáváme:

$$ \large R(t) = e^{-\lambda_{s} \cdot t} $$

In [5]:
import math
def computer_running_probability(computer_failure_intensity, t):
    return math.exp(-computer_failure_intensity * t)

Pravděpodobnost, že počítač zrealizuje službu trvající $ T $ hodin:
$$ serviceCompletedProbability = R(T) $$

> Dále vypočítejte uvedené tři výsledky též numericky pro K  = 10, N  = 10000, T  = 10 hod.

In [6]:
# 10 components failed
K = 10

In [7]:
# computer is made of 10000 components
N = 10000

In [8]:
# service takes 10 hours
T = 10

In [9]:
component_failure_intensity_hours = single_component_failure_intensity_hours(K, N)
component_failure_intensity_hours

1.1415525114155251e-07

Intenzita poruchy jedné komponenty $λ$:
$$ λ = N \cdot \frac{\frac{K}{N}}{365 \cdot 24} = \frac{\frac{10}{10000}}{365 \cdot 24} = 1.14155 \cdot 10^{-7} \ \mathrm{hod^{-1}}$$

In [10]:
computer_failure_intensity_hours = computer_failure_intensity(N, component_failure_intensity_hours)
computer_failure_intensity_hours

0.001141552511415525

Intenzita poruchy celého systému $\lambda_{s}$:
$$ \lambda_{s} = N \cdot \lambda = 0.00114155\ \mathrm{hod^{-1}}$$

In [11]:
computer_running_probability(computer_failure_intensity_hours, T)

0.9886493847644267

Pravděpododobnost, že počítač zrealizuje službu trvající $ T = 10 $ hodin:
$$ \large  R(t = T = 10) = e^{-\lambda_{s} \cdot t} = e^{-0.00114155 \cdot 10} = 0.988649 $$,
tedy **98.8649 %**.